sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3344,7 +3344,7 @@ pi_result piextMemCreateWithNativeHandle(pi_native_handle NativeHandle,
     pi_platform Plt = Context->getPlatform();
     std::unique_lock<pi_shared_mutex> ContextsLock(Plt->ContextsMutex,
                                                    std::defer_lock);
-    if (IndirectAccessTrackingEnabled) {
+    if (IndirectAccessTrackingEnabled && ownNativeHandle) {
```

Comment:
Please add a source code comment why we check the ownership

Reply:
Added source code comment.